core/lib/dal/src/tee_proof_generation_dal.rs

```diff
@@ -16,6 +17,19 @@ pub struct TeeProofGenerationDal<'a, 'c> {
     pub(crate) storage: &'a mut Connection<'c, Core>,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TeeProof {
+    // signature generated within the TEE enclave, using the privkey corresponding to the pubkey
+    pub signature: Option<Vec<u8>>,
+    // pubkey used for signature verification; each key pair is attested by the TEE attestation
+    // stored in the db
+    pub pubkey: Option<Vec<u8>>,
+    // data that was signed
+    pub proof: Option<Vec<u8>>,
+    // type of TEE used for attestation
+    pub tee_type: Option<TeeType>,
+}
```

Comment:
This is almost identical to this `struct`:
https://github.com/matter-labs/zksync-era/blob/f42b26027cf864a08b775fa2ed04adcddf5311a9/core/lib/prover_interface/src/outputs.rs#L16-L28
(with a minor change as I needed to introduce `Option`s here)

I don't have a clear idea on how to tidy it up at this point. I had an idea to move `struct TeeProof` definition to
https://github.com/matter-labs/zksync-era/blob/main/core/lib/basic_types/src/tee_types.rs
but then I would need to move this `impl` as well:
https://github.com/matter-labs/zksync-era/blob/f42b26027cf864a08b775fa2ed04adcddf5311a9/core/lib/prover_interface/src/outputs.rs#L60-L69
This would force me to add [`zksync_object_store`](https://github.com/matter-labs/zksync-era/blob/5c932f49fe5a5899df0e7b6a9fe55756d4559800/core/lib/object_store/Cargo.toml#L2C9-L2C28) dependency to the [`zksync_basic_types`](https://github.com/matter-labs/zksync-era/blob/5c932f49fe5a5899df0e7b6a9fe55756d4559800/core/lib/basic_types/Cargo.toml#L2C9-L2C27) crate, which is unexpected and could potentially introduce a circular dependency.

Any suggestions are welcome, @popzxc.

Reply:
You need to define `StorageTeeProof` structure in the `models` module. Take a look at other examples.